### ISSUE TITLE
add alternative name for "node-args" for other backends

### ISSUE THREAD

**Issue by joprice** [purescript/spago#709]
I was surprised to find that `node-args` works for any backend: https://github.com/andyarvanitis/purescript-native/issues/67. Could there be a new alias for the flag that doesn't mention node? Looking at the code, it seems that it's already represented by the type `BackendArg`, just not exposed that way.

**Comment by f-f** [comment]
@joprice definitely! As you suggested we could just add a `backend-args` flag that works alongside `node-args`. This should be straightforward to implement in our CLI code, so I'll mark it as up for grabs

**Comment by joprice** [comment]
I'd like to try this one myself.

**Comment by joprice** [comment]
I started by building the project and running against an existing test repo of mine, and it looks like there was a backward incompatible change in dhall parsing:

```
spago:
↳ ./spago.dhall
  ↳ ./../packages.dhall

Error: Invalid input

/Users/josephprice/dev/purescript-native-test/main/../packages.dhall:131:7:
    |
131 |       with parser = ./parser/spago.dhall as Location
    |       ^
unexpected 'w'
expecting ->, :, end of input, or whitespace


21│                  ../packages.dhall

/Users/josephprice/dev/purescript-native-test/main/spago.dhall:21:18
```

```dhall

  let upstream =
        https://github.com/purescript/package-sets/releases/download/psc-0.13.8-20200922/packages.dhall sha256:5edc9af74593eab8834d7e324e5868a3d258bbab75c5531d2eb770d4324a2900

  let overrides = {=}

  let additions = {=}

  in      upstream
      //  overrides
      //  additions
        with parser = ./parser/spago.dhall as Location
        with main = ./main/spago.dhall as Location
```

Adding an extra let for the merged upstream overrides and additions fixes it.

**Comment by f-f** [comment]
Fixed in #710
